### ISSUE TITLE
🐛 Parsing error in Astro files

### ISSUE THREAD

**Issue by arv-anshul** [biomejs/biome#8950]
### Environment information

<details>

```bash
CLI:
  Version:                      2.3.12
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v25.4.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.3.4

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    true

Workspace:
  Open Documents:               0
```

</details>

### What happened?

Unexpected error, if the `/* */` comment contains single quote `'` once.

<img width="1152" height="900" alt="Image" src="https://github.com/user-attachments/assets/869a1cad-4ce8-4573-b391-cba44ea5ba15" />

### Expected result

When the single is removed or introduce another the error's gone.

Like:

1. Remove that single quote.

<img width="391" height="420" alt="Image" src="https://github.com/user-attachments/assets/8050187a-a7e2-4d9a-a94a-4661f57f9d23" />

2. Introduce another single quote.

<img width="398" height="429" alt="Image" src="https://github.com/user-attachments/assets/d4504317-e92a-4d4b-9494-935fb601495c" />

3. Introduce another single quote in another **MULTILINE COMMENT**.

<img width="822" height="606" alt="Image" src="https://github.com/user-attachments/assets/2f637d1a-c907-4253-9f43-0f9393f19614" />

> [!note]
>
> I have also encounter couple of other method which will lead to this. If you ask, I can send those too.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
Please provide reproductions in one of our preferred formats. It helps us stay organized.

**Comment by arv-anshul** [comment]
Sorry!

Reproduced in [Biome playground](https://biomejs.dev/playground/?indentScriptAndStyle=true&enabledLinting=false&code=LQAtAC0ACgAvACoAIABJAG4AdAByAG8AZAB1AGMAZQAgAGEAbgBvAHQAaABlAHIAIABsAGkAawBlADoALgAnAC4ALwAvAC8AIAA9AD0APgAgAFQAaABpAHMAIABkAG8AZQBzACAAbgBvAHQAIABzAG8AbAB2AGUAIAB0AGgAZQAgAGUAcgByAG8AcgAKAAoALwAqAAoAIAAqACAASQBuAHQAcgBvAGQAdQBjAGUAIABhAG4AbwB0AGgAZQByACAAYgB1AHQAIABpAG4ALQBtAHUAbAB0AGkAbABpAG4AZQAgAGMAbwBtAG0AZQBuAHQAIABsAGkAawBlADoAIAA8AGEAZABkACAAMQAgAHMAaQBuAGcAbABlACAAcQB1AG8AdABlAD4AIAA8AC0ALQAKACAAKgAvACAALwAvACAAPQA9AD4AIABUAGgAaQBzACAAcwBvAGwAdgBlAHMAIAB0AGgAZQAgAGUAcgByAG8AcgAKAAoAaQBuAHQAZQByAGYAYQBjAGUAIABQAHIAbwBwAHMAIAB7AAoAIAAgAC8AKgAqAAoAIAAgACoAIABBAG4AIABvAHAAdABpAG8AbgBhAGwAIABwAGEAZwBlACcAcwAgAHAAcgBvAHAALgAKACAAIAAqAC8ACgAgACAAbwBwAHQAaQBvAG4AYQBsAD8AOgAgAHMAdAByAGkAbgBnADsACgB9AAoACgBjAG8AbgBzAHQAIAB0AGkAdABsAGUAIAA9ACAAIgBIAGUAbABsAG8AIABXAG8AcgBsAGQAIQAiADsACgAKAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoACIAaABpACIAKQA7AAoACgAtAC0ALQAKACAAIAAKADwAZABpAHYAPgB7AHQAaQB0AGwAZQB9ADwALwBkAGkAdgA%2BAA%3D%3D&language=astro).

**Comment by astrochemx** [comment]
Isn't this the same issue as [#8882](https://github.com/biomejs/biome/issues/8882)?
